Kernel/FileSystem/BlockBasedFileSystem.cpp

```diff
@@ -116,6 +124,15 @@ BlockBasedFileSystem::BlockBasedFileSystem(OpenFileDescription& file_description
 
 BlockBasedFileSystem::~BlockBasedFileSystem() = default;
 
+void BlockBasedFileSystem::remove_disk_cache_before_last_unmount()
+{
+    VERIFY(m_lock.is_locked());
```

Comment:
Why do we need to hold the per-filesystem lock here when we are taking a narrower lock on the cache?

Reply:
This is just to ensure this happens on certain operations, like in the last mount removal handling.

Reply:
Gotcha. I wonder if in the future we can keep narrowing the locks further.